src/net/inv.rs

```diff
@@ -2249,7 +2249,7 @@ impl PeerNetwork {
         &mut self,
         sortdb: &SortitionDB,
         ibd: bool,
-    ) -> Result<(bool, bool, Vec<NeighborKey>, Vec<NeighborKey>), net_error> {
+    ) -> (bool, bool, Vec<NeighborKey>, Vec<NeighborKey>) {
         PeerNetwork::with_inv_state(self, |network, inv_state| {
```

Comment:
Is there any path in this anonymous function that returns `Err`? I'd be more comfortable with the `expect()` at the end of this call if `with_inv_state`'s signature looked more like:

```
    pub fn with_inv_state<F, R>(network: &mut PeerNetwork, handler: F) -> Result<R, net_error>
    where
        F: FnOnce(&mut PeerNetwork, &mut InvState) -> R,
```

That way, the `.expect()` is explicitly only unwrapping a failure in the outer method (so, it's just a failure in opening the inv state).

Reply:
Sure, changed it to that signature and refactored the needful.